kernel/ixgbe/src/lib.rs

```diff
@@ -1213,29 +1302,29 @@ pub enum FilterProtocol {
 }
 
 /// A helper function to poll the nic receive queues (only for testing purposes).
-fn rx_poll_mq(qid: usize) -> Result<ReceivedFrame, &'static str> {
-    let nic_ref = get_ixgbe_nic().ok_or("ixgbe nic not initialized")?;
+fn rx_poll_mq(qid: usize, nic_id: PciLocation) -> Result<ReceivedFrame, &'static str> {
+    let nic_ref = get_ixgbe_nic(nic_id).ok_or("ixgbe nic not initialized")?;
```

Comment:
should this error message be changed to reflect that the requested nic_id didn't exist (or wasn't inited)?

Reply:
updated get_ixgbe_nic function to return a result that covers the two possibilities